core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -82,11 +83,17 @@
 
     private final BigArrays bigArrays;
     private final ScriptService scriptService;
+    private final Client client;
 
-    public SearchPhaseController(Settings settings, BigArrays bigArrays, ScriptService scriptService) {
+    public SearchPhaseController(Settings settings, BigArrays bigArrays, ScriptService scriptService, Client client) {
         super(settings);
         this.bigArrays = bigArrays;
         this.scriptService = scriptService;
+        this.client = client;
+    }
+
+    public Client getClient() {
```

Comment:
Right, we don't need the client here.
Does https://github.com/elastic/elasticsearch/pull/23053/commits/2c34484c607926d39ec6f0152e84f6185eaa7fed works for you ?